### PR TITLE
Generic Reification Support

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Add support for generating schemas for generic types. For example:
```kotlin
get<genericPets>("all".responds(ok<Model<PetModel>>())) {
    call.respond(data)
}
```
The above should generate a schema named `ModelOfPetModel` and a schema for `PetModel`.

### CODE REVIEW

File: ktor-swagger/ktor-swagger.gradle.kts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 
 dependencies {
     implementation(group = "org.webjars", name = "swagger-ui", version = "3.1.5")
+    implementation(group = "org.apache.commons", name = "commons-lang3", version = "3.7")
```
Comment (JLLeitschuh) [inline ktor-swagger/ktor-swagger.gradle.kts]: Not a fan of having to add this dependency but `TypeUtils.parameterize` simplified this code significantly.

File: ktor-swagger/src/main/kotlin/de/nielsfalk/playground/ktor/swagger/Swagger.kt [hunk 2]
```diff
@@ -219,24 +232,136 @@ val propertyTypes = mapOf(
...
+fun KType.toModelProperty(reifiedType: Type?): Pair<Property, Collection<TypeInfo>> =
+    resolveTypeInfo(reifiedType).let { typeInfo ->
+        val type = typeInfo?.type ?: classifier as KClass<*>
+        type.toModelProperty(this, typeInfo?.reifiedType as? ParameterizedType)
+    }
+
+internal fun <T, R> KProperty1<T, R>.returnTypeInfo(reifiedType: Type?): TypeInfo =
+    returnType.resolveTypeInfo(reifiedType)!!
+
+internal fun KType.resolveTypeInfo(reifiedType: Type?): TypeInfo? {
```
Comment (Octogonapus) [inline ktor-swagger/src/main/kotlin/de/nielsfalk/playground/ktor/swagger/Swagger.kt]: Does this need to be nullable? It looks like the only nullable case is `KClass<*>` which is then immediately non-null asserted.

File: ktor-swagger/src/main/kotlin/de/nielsfalk/playground/ktor/swagger/SwaggerParameterizedType.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Create a parameterized type instance.
+ * @param raw the raw class to create a parameterized type instance for
+ * @param typeArguments the mapping used for parameterization
+ * @return [ParameterizedType]
+ */
+internal fun parameterize(raw: Class<*>, vararg typeArguments: Type): ParameterizedType {
+    val useOwner: Type?
+    if (raw.enclosingClass == null) {
```
Comment (Octogonapus) [inline ktor-swagger/src/main/kotlin/de/nielsfalk/playground/ktor/swagger/SwaggerParameterizedType.kt]: Maybe this if should be an expression?

Comment (JLLeitschuh) [reply ktor-swagger/src/main/kotlin/de/nielsfalk/playground/ktor/swagger/SwaggerParameterizedType.kt]: 👍

File: ktor-swagger/src/main/kotlin/de/nielsfalk/playground/ktor/swagger/Swagger.kt (hunk 2 above)
Comment (JLLeitschuh) [reply ktor-swagger/src/main/kotlin/de/nielsfalk/playground/ktor/swagger/Swagger.kt]: I answered offline, but for documentation purposes, yes. Null is used.

**Pull request merged**